scio-core/src/main/scala/com/spotify/scio/ScioResult.scala

```diff
@@ -64,18 +65,30 @@ abstract class ScioResult private[scio] (val internal: PipelineResult) {
   /** Get metrics of the finished pipeline. */
   def getMetrics: Metrics
 
-  /** Wait until the pipeline finishes. */
-  def waitUntilFinish(duration: Duration = Duration.Inf): ScioResult = {
-    Await.ready(finalState, duration)
+  /** Wait until the pipeline finishes. If timeout duration is exceeded and `doCancelJob` is set,
+    * cancel the internal [[PipelineResult]]. */
+  def waitUntilFinish(duration: Duration = Duration.Inf, doCancelJob: Boolean = false):
```

Comment:
@ClaireMcGinty 🎨 this is a nit: i think I would prefer `cancelJob` instead of `doCancelJob`